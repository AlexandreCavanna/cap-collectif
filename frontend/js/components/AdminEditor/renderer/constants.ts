// All atomic block entity types
export const IFRAME = 'IFRAME'
export const IMAGE = 'IMAGE'
export const HR = 'HR'
export const LINK = 'LINK'
export const TABLE = 'TABLE'
// WIP
export const COLORS = [
  '#000000',
  '#FFFFFF',
  '#888888',
  '#AAAAAA',
  '#EEEEEE',
  '#880000',
  '#CC0000',
  '#FF0000',
  '#FFCCCC',
  '#FF8800',
  '#FFCC00',
  '#FFFF00',
  '#CCFF00',
  '#88FF00',
  '#008800',
  '#00CC00',
  '#00CC88',
  '#00CCCC',
  '#CCEEFF',
  '#00CCFF',
  '#0088FF',
  '#0000FF',
  '#8800FF',
  '#CC00CC',
  '#CC0088',
]
export const FONTS = [
  'Arial',
  'Arial Black',
  'Comic Sans MS',
  'Courier New',
  'Helvetica Neue',
  'Helvetica',
  'Impact',
  'Lucida Grande',
  'Tahoma',
  'Times New Roman',
  'Verdana',
]
export const FONT_SIZES = [8, 9, 10, 11, 12, 14, 18, 24, 36]
