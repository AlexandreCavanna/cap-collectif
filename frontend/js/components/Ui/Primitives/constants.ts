import { SPACES_SCALES } from '~/styles/theme/base'

export const FontWeight = {
  Hairline: 'hairline',
  Thin: 'thin',
  Light: 'light',
  Normal: 'normal',
  Medium: 'medium',
  Semibold: 'semibold',
  Bold: 'bold',
  Extrabold: 'extrabold',
  Black: 'black',
}
export const LineHeight = {
  SM: 'sm',
  S: 's',
  Base: 'base',
  Normal: 'normal',
  M: 'm',
  L: 'l',
  XL: 'xl',
}
export const LetterSpacing: {
  Tighter: 'tighter'
  Tight: 'tight'
  Normal: 'normal'
  Wide: 'wide'
  Wider: 'wider'
  Widest: 'widest'
} = {
  Tighter: 'tighter',
  Tight: 'tight',
  Normal: 'normal',
  Wide: 'wide',
  Wider: 'wider',
  Widest: 'widest',
}
export const Spacing = {
  None: Number(SPACES_SCALES[0]),
  Xs2: Number(SPACES_SCALES[1].replace('rem', '')) * 16,
  Xs: Number(SPACES_SCALES[2].replace('rem', '')) * 16,
  Base: Number(SPACES_SCALES[3].replace('rem', '')) * 16,
  Medium: Number(SPACES_SCALES[4].replace('rem', '')) * 16,
  Large: Number(SPACES_SCALES[5].replace('rem', '')) * 16,
  Xl: Number(SPACES_SCALES[6].replace('rem', '')) * 16,
  Xl2: Number(SPACES_SCALES[7].replace('rem', '')) * 16,
  Xl3: Number(SPACES_SCALES[8].replace('rem', '')) * 16,
}
