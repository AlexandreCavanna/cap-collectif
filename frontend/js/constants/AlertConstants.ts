export const UPDATE_ALERT = 'UPDATE_ALERT'
export const TYPE_ALERT: {
  SUCCESS: 'SUCCESS'
  ERROR: 'ERROR'
} = {
  SUCCESS: 'SUCCESS',
  ERROR: 'ERROR',
}
