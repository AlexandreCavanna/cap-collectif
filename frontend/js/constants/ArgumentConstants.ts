export const COMMENT_SYSTEM_NONE = 0
export const COMMENT_SYSTEM_SIMPLE = 1
export const COMMENT_SYSTEM_BOTH = 2
