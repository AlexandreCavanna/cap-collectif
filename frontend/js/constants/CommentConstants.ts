export const CREATE_COMMENT_SUCCESS = 'CREATE_COMMENT_SUCCESS'
export const CREATE_COMMENT_FAILURE = 'CREATE_COMMENT_FAILURE'
export const RECEIVE_COMMENTS = 'RECEIVE_COMMENTS'
