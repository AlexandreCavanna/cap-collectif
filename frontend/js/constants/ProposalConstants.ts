export const VOTE_TYPE_DISABLED = 0
export const VOTE_TYPE_SIMPLE = 1
export const VOTE_TYPE_BUDGET = 2
export const PROPOSAL_VOTES_TO_SHOW = 6
export const PROPOSAL_FOLLOWERS_TO_SHOW = 32
// === order
export const PROPOSAL_ORDER_RANDOM = 'random'
export const PROPOSAL_ORDER_LAST = 'last'
export const PROPOSAL_ORDER_OLD = 'old'
export const PROPOSAL_ORDER_COMMENTS = 'comments'
export const PROPOSAL_ORDER_COST_EXPENSIVE = 'expensive'
export const PROPOSAL_ORDER_COST_CHEAP = 'cheap'
export const PROPOSAL_AVAILABLE_ORDERS = [PROPOSAL_ORDER_RANDOM, PROPOSAL_ORDER_LAST, PROPOSAL_ORDER_OLD]
