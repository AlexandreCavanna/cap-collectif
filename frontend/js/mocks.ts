/* eslint-env jest */
import type { IntlShape, MessageDescriptor } from 'react-intl'
import 'react-intl'
import type { RelayPaginationProp, RelayRefetchProp } from 'react-relay'

export const $refType: any = null
export const $fragmentRefs: any = null
export const intlMock: IntlShape = {
  locale: 'fr-FR',
  formats: {},
  messages: {},
  now: () => 0,
  formatHTMLMessage: (message: MessageDescriptor) => String(message),
  formatPlural: (message: string) => String(message),
  formatNumber: (message: string) => String(message),
  formatRelative: (message: string) => String(message),
  formatTime: (message: string) => String(message),
  formatDate: (message: string) => String(message),
  formatMessage: (message: MessageDescriptor) => String(message.id),
}
export const formMock: ReduxFormFormProps = {
  anyTouched: false,
  array: {
    insert: jest.fn(),
    move: jest.fn(),
    pop: jest.fn(),
    push: jest.fn(),
    remove: jest.fn(),
    removeAll: jest.fn(),
    shift: jest.fn(),
    splice: jest.fn(),
    swap: jest.fn(),
    unshift: jest.fn(),
  },
  asyncValidate: jest.fn(),
  asyncValidating: false,
  autofill: jest.fn(),
  blur: jest.fn(),
  change: jest.fn(),
  clearAsyncError: jest.fn(),
  clearSubmit: jest.fn(),
  destroy: jest.fn(),
  dirty: false,
  dispatch: jest.fn(),
  error: null,
  form: 'formName',
  handleSubmit: jest.fn(),
  initialize: jest.fn(),
  initialized: true,
  // initialValues: {},
  invalid: false,
  pristine: true,
  reset: jest.fn(),
  submitting: false,
  submitFailed: false,
  submitSucceeded: false,
  touch: jest.fn(),
  untouch: jest.fn(),
  updateSyncErrors: jest.fn(),
  valid: true,
  warning: null,
}
const environment = {
  UNSTABLE_getDefaultRenderPolicy: jest.fn(),
  __log: jest.fn(),
  getOperationTracker: jest.fn(),
  isRequestActive: jest.fn(),
  isServer: jest.fn(),
  options: jest.fn(),
  applyMutation: jest.fn(),
  sendMutation: jest.fn(),
  lookup: jest.fn(),
  sendQuery: jest.fn(),
  subscribe: jest.fn(),
  streamQuery: jest.fn(),
  retain: jest.fn(),
  unstable_internal: {
    areEqualSelectors: jest.fn(),
    createFragmentSpecResolver: jest.fn(),
    createOperationDescriptor: jest.fn(),
    getDataIDsFromFragment: jest.fn(),
    getDataIDsFromObject: jest.fn(),
    getFragment: jest.fn(),
    getPluralSelector: jest.fn(),
    getRequest: jest.fn(),
    getSelector: jest.fn(),
    getSelectorsFromObject: jest.fn(),
    getSingularSelector: jest.fn(),
    getVariablesFromFragment: jest.fn(),
    getVariablesFromObject: jest.fn(),
    getVariablesFromPluralFragment: jest.fn(),
    getVariablesFromSingularFragment: jest.fn(),
    isFragment: jest.fn(),
    isRequest: jest.fn(),
  },
  applyUpdate: jest.fn(),
  check: jest.fn(),
  commitPayload: jest.fn(),
  commitUpdate: jest.fn(),
  execute: jest.fn(),
  executeMutation: jest.fn(),
  executeWithSource: jest.fn(),
  getNetwork: jest.fn(),
  getStore: jest.fn(),
  areEqualSelectors: jest.fn(),
  createFragmentSpecResolver: jest.fn(),
  createOperationDescriptor: jest.fn(),
  getDataIDsFromFragment: jest.fn(),
  getDataIDsFromObject: jest.fn(),
  getFragment: jest.fn(),
  getPluralSelector: jest.fn(),
  getRequest: jest.fn(),
  getSelector: jest.fn(),
  getSelectorsFromObject: jest.fn(),
  getSingularSelector: jest.fn(),
  getVariablesFromFragment: jest.fn(),
  getVariablesFromObject: jest.fn(),
  getVariablesFromPluralFragment: jest.fn(),
  requiredFieldLogger: jest.fn(),
  revertUpdate: jest.fn(),
  replaceUpdate: jest.fn(),
}
export const relayPaginationMock: RelayPaginationProp = {
  environment,
  hasMore: () => false,
  isLoading: () => false,
  loadMore: jest.fn(),
  refetchConnection: jest.fn(),
}
export const relayRefetchMock: RelayRefetchProp = {
  environment,
  refetch: jest.fn(),
}
export const googleAddressMock = {
  lat: 43.6424564,
  lng: -79.3755156,
  json: '[{"formatted_address":"10 Yonge St, Toronto, ON M5E 1R4, Canada","geometry":{"location":{"lat":43.6424564,"lng":-79.3755156},"location_type":"ROOFTOP","viewport":{"south":43.6411074197085,"west":-79.37686458029151,"north":43.6438053802915,"east":-79.37416661970849}},"types":["street_address"],"address_components":[{"long_name":"10","short_name":"10","types":["street_number"]},{"long_name":"Yonge Street","short_name":"Yonge St","types":["route"]},{"long_name":"Old Toronto","short_name":"Old Toronto","types":["political","sublocality","sublocality_level_1"]},{"long_name":"Toronto","short_name":"Toronto","types":["locality","political"]},{"long_name":"Toronto Division","short_name":"Toronto Division","types":["administrative_area_level_2","political"]},{"long_name":"Ontario","short_name":"ON","types":["administrative_area_level_1","political"]},{"long_name":"Canada","short_name":"CA","types":["country","political"]},{"long_name":"M5E 1R4","short_name":"M5E 1R4","types":["postal_code"]}],"place_id":"ChIJrbgtxSvL1IkR2xGe7JZyuw0","plus_code":{"compound_code":"JJRF+XQ Toronto, ON, Canada","global_code":"87M2JJRF+XQ"}}]',
}
