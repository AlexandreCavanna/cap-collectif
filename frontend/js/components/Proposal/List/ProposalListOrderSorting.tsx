import * as React from 'react'
import type { IntlShape } from 'react-intl'
import { injectIntl } from 'react-intl'
import { connect } from 'react-redux'
import LocalStorage from '../../../services/LocalStorageService'
import { changeOrder } from '~/redux/modules/proposal'
import { PROPOSAL_AVAILABLE_ORDERS } from '~/constants/ProposalConstants'
import type { Dispatch, State } from '~/types'
import Select from '../../Ui/Form/Select/Select'
import SelectOption from '../../Ui/Form/Select/SelectOption'

type Props = {
  orderByPoints?: boolean
  orderByVotes?: boolean
  orderByComments?: boolean
  orderByCost?: boolean
  dispatch: Dispatch
  order?: string
  defaultSort?: string | null | undefined
  objectType: string
  stepId?: string
  intl: IntlShape
  canDisplayBallot: boolean
}
type ComponentState = {
  displayedOrders: Array<string>
}
export class ProposalListOrderSorting extends React.Component<Props, ComponentState> {
  static defaultProps = {
    orderByPoints: false,
    orderByVotes: false,
    orderByComments: false,
    orderByCost: false,
  }

  constructor(props: Props) {
    super(props)
    this.state = {
      displayedOrders: PROPOSAL_AVAILABLE_ORDERS.concat(props.orderByComments ? ['comments'] : [])
        .concat(props.orderByCost ? ['expensive', 'cheap'] : [])
        .concat(props.orderByVotes && props.canDisplayBallot ? ['votes', 'least-votes'] : [])
        .concat(props.orderByPoints && props.canDisplayBallot ? ['points', 'least-points'] : []),
    }
  }

  componentDidMount() {
    const { dispatch, defaultSort, stepId, order } = this.props
    const savedSort = LocalStorage.get('proposal.orderByStep')

    if (!savedSort || !Object.prototype.hasOwnProperty.call(savedSort, stepId)) {
      if (defaultSort && !order) {
        dispatch(changeOrder(defaultSort))
      }
    } else {
      dispatch(changeOrder(savedSort[stepId]))
    }
  }

  render() {
    const { order, dispatch, intl, objectType } = this.props
    const { displayedOrders } = this.state
    const isOpinion = objectType === 'OPINION'
    const filterPrefix = isOpinion ? '' : 'f_'
    return (
      <div>
        <Select
          label={
            order
              ? intl.formatMessage({
                  id: `global.filter_${filterPrefix}${order}`,
                })
              : intl.formatMessage({
                  id: 'global.filter',
                })
          }
          id="proposal-filter-sorting-button"
        >
          {displayedOrders.map(choice => (
            <SelectOption
              key={choice}
              onClick={e => dispatch(changeOrder(e.currentTarget.value))}
              isSelected={choice === order}
              value={choice}
            >
              {intl.formatMessage({
                id: `global.filter_${filterPrefix}${choice}`,
              })}
            </SelectOption>
          ))}
        </Select>
      </div>
    )
  }
}

const mapStateToProps = (state: State) => ({
  order: state.proposal.order,
  stepId: state.project.currentProjectStepById || null,
})

const container = injectIntl(ProposalListOrderSorting)
export default connect<any, any>(mapStateToProps)(container)
