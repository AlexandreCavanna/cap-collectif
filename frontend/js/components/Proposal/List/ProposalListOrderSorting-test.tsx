/* eslint-env jest */
import React from 'react'
import { shallow } from 'enzyme'
import { ProposalListOrderSorting } from './ProposalListOrderSorting'
import { PROPOSAL_ORDER_RANDOM } from '../../../constants/ProposalConstants'
import { intlMock } from '../../../mocks'

const props = {
  orderByVotes: false,
  dispatch: jest.fn(),
  order: PROPOSAL_ORDER_RANDOM,
  defaultSort: PROPOSAL_ORDER_RANDOM,
  stepId: 'selectionstep1',
  objectType: 'PROPOSAL',
  intl: intlMock,
  canDisplayBallot: true,
}
describe('<ProposalListOrderSorting />', () => {
  it('should render the list order sorting component', () => {
    const wrapper = shallow(<ProposalListOrderSorting {...props} />)
    expect(wrapper).toMatchSnapshot()
  })
  it('should render the list order sorting component without vote sort', () => {
    const wrapper = shallow(<ProposalListOrderSorting {...props} canDisplayBallot={false} />)
    expect(wrapper).toMatchSnapshot()
  })
})
