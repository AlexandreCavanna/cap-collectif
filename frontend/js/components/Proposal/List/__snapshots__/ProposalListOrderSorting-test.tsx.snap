// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProposalListOrderSorting /> should render the list order sorting component 1`] = `
<div>
  <Select
    id="proposal-filter-sorting-button"
    label="global.filter_f_random"
  >
    <SelectOption
      isSelected={true}
      key="random"
      onClick={[Function]}
      value="random"
    >
      global.filter_f_random
    </SelectOption>
    <SelectOption
      isSelected={false}
      key="last"
      onClick={[Function]}
      value="last"
    >
      global.filter_f_last
    </SelectOption>
    <SelectOption
      isSelected={false}
      key="old"
      onClick={[Function]}
      value="old"
    >
      global.filter_f_old
    </SelectOption>
  </Select>
</div>
`;

exports[`<ProposalListOrderSorting /> should render the list order sorting component without vote sort 1`] = `
<div>
  <Select
    id="proposal-filter-sorting-button"
    label="global.filter_f_random"
  >
    <SelectOption
      isSelected={true}
      key="random"
      onClick={[Function]}
      value="random"
    >
      global.filter_f_random
    </SelectOption>
    <SelectOption
      isSelected={false}
      key="last"
      onClick={[Function]}
      value="last"
    >
      global.filter_f_last
    </SelectOption>
    <SelectOption
      isSelected={false}
      key="old"
      onClick={[Function]}
      value="old"
    >
      global.filter_f_old
    </SelectOption>
  </Select>
</div>
`;
